nft-simple/src/nft_core.rs

```diff
@@ -133,23 +145,49 @@ impl NonFungibleTokenCore for Contract {
     }
 
     #[payable]
-    fn nft_approve_account_id(&mut self, token_id: TokenId, account_id: ValidAccountId) -> bool {
-        assert_one_yocto();
+    fn nft_approve(
+        &mut self,
+        token_id: TokenId,
+        account_id: ValidAccountId,
+        msg: Option<String>,
+    ) -> bool {
+        let mut deposit = env::attached_deposit();
+        let account_id: AccountId = account_id.into();
+
+        let storage_required = bytes_for_approved_account_id(&account_id);
+        assert!(deposit >= storage_required as u128, "Deposit doesn't cover storage of account_id: {}", account_id.clone());
+
         let mut token = self.tokens_by_id.get(&token_id).expect("Token not found");
         assert_eq!(&env::predecessor_account_id(), &token.owner_id);
-        let account_id: AccountId = account_id.into();
-        let storage_used = bytes_for_approved_account_id(&account_id);
-        if token.approved_account_ids.insert(account_id) {
-            deposit_refund(storage_used);
+
+        let mut result = false;
+        if token.approved_account_ids.insert(account_id.clone()) {
             self.tokens_by_id.insert(&token_id, &token);
-            true
-        } else {
-            false
+            deposit -= storage_required as u128;
+            result = true;
         }
```

Comment:
We don't need result at all. Since if it didn't panic, then we added it and called the receiver.